src/widgets/index.tsx

```diff
@@ -29,32 +30,44 @@ type IWidgetMap = {
     params: Extract<IWidget, {component: key}> & {
       children?: ReactNode[];
       deep?: boolean;
+      id?: string;
     },
   ) => ReactNode;
 };
 
 const WidgetMap: IWidgetMap = {
   Transactions: params => (
-    <TransactionsWidgetWrapper key={generateUUID()} {...params} />
+    <TransactionsWidgetWrapper key={params.id ?? generateUUID()} {...params} />
```

Comment:
Lets convert it to
```suggestion
    <TransactionsWidgetWrapper key={generateUUID(params)} {...params} />
```

And we will have incapsulated logic about UUID generation in `generateUUID` function